src/btree/bt_split.c

```diff
@@ -698,7 +698,10 @@ __split_parent(WT_SESSION_IMPL *session, WT_REF *ref, WT_REF **ref_new, uint32_t
              * unless we update the internal page's start recno on the fly and restart the search,
              * which seems like asking for trouble.) Don't discard any ref has the prefetch flag,
              * the prefetch thread would crash if it sees a freed ref.
-             */
+             */      
+            if (next_ref->ref_changes != 0)
```

Comment:
We should only continue if the ref is in WT_REF_DELETED state. So we can merge this check with the following if check.